Bottom: 33f25ea0ddc9959e9dc86b9e7a28efb00600104f
Top:    76566d2e62d0dc3c905d7466a8dd7df6b6b3d14f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-13 05:23:52 +0100

Try optimising polygon operations by keeping an rtree with their contours

InsertHoles() already creates an rtree of contours to do its internal
processing, and it is possible that intersect() could be sped up by
its existance as well.




---

diff --git a/src/polyarea.h b/src/polyarea.h
index 42acc1a..819b074 100644
--- a/src/polyarea.h
+++ b/src/polyarea.h
@@ -128,6 +128,7 @@ struct POLYAREA
 {
     POLYAREA *f, *b;
     PLINE *contours;
+    void *contour_tree;
 };
 
 BOOLp poly_M_Copy0(POLYAREA ** dst, const POLYAREA * srcfst);
